File: Examples/ClassDiagram/Dart/podcast.dart
import 'media_item.dart';
import 'playable.dart';

/// One episode of a podcast.
class Podcast extends MediaItem {
  Podcast(String title, double duration, this.host, this.episodeNumber)
      : super(title, duration, Genre.spokenWord);

  final String host;
  final int episodeNumber;

  @override
  void play() {
    print('🎙 Episode $episodeNumber: $title');
  }
}

File: Examples/ClassDiagram/TypeScript/Podcast.ts
import { MediaItem } from "./MediaItem";
import { Genre } from "./Playable";

/** One episode of a podcast. */
export class Podcast extends MediaItem {
    constructor(
        title: string,
        duration: number,
        readonly host: string,
        readonly episodeNumber: number,
    ) {
        super(title, duration, Genre.SpokenWord);
    }

    override play(): void {
        console.log(`🎙 Episode ${this.episodeNumber}: ${this.title}`);
    }
}
